packages/orm/src/EntityManager.ts

```diff
@@ -1934,15 +1934,16 @@ function recalcSynchronousDerivedFields(todos: Record<string, Todo>) {
     [...new Set(entities.map(getMetadata))].map((m) => {
       return [
         m,
-        Object.values(m.fields)
-          .filter((f) => f.kind === "primitive" && f.derived === "sync")
+        Object.values(m.allFields)
+          .filter((f) => (f.kind === "primitive" || f.kind === "enum") && f.derived === "sync")
```

Comment:
@stephenh - I'm pretty sure the lack of an enum check here was also a bug. I could add tests if desired.